data race in unittests

![image](https://github.com/openshift-pipelines/pipelines-as-code/assets/98980/2b22b3cf-3232-442c-9675-2b818af16c1f)

```
[go-ninja-go : unittest] === RUN   TestHandleEvent
[go-ninja-go : unittest] === RUN   TestHandleEvent/get_http_call
[go-ninja-go : unittest] === RUN   TestHandleEvent/invalid_json_body
[go-ninja-go : unittest] === RUN   TestHandleEvent/invalid_json_body_only_when_payload_has_been_set
[go-ninja-go : unittest] === RUN   TestHandleEvent/valid_event
[go-ninja-go : unittest] === RUN   TestHandleEvent/skip_event
[go-ninja-go : unittest] ==================
[go-ninja-go : unittest] WARNING: DATA RACE
[go-ninja-go : unittest] Write at 0x00c0006b1800 by goroutine 44:
[go-ninja-go : unittest]   github.com/openshift-pipelines/pipelines-as-code/pkg/adapter.listener.handleEvent.func1()
[go-ninja-go : unittest]       /workspace/source/pkg/adapter/adapter.go:111 +0x2a4
[go-ninja-go : unittest]   net/http.HandlerFunc.ServeHTTP()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:2122 +0x4d
[go-ninja-go : unittest]   net/http.serverHandler.ServeHTTP()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:2936 +0x682
[go-ninja-go : unittest]   net/http.(*conn).serve()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:1995 +0xbd4
[go-ninja-go : unittest]   net/http.(*Server).Serve.func3()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:3089 +0x58
[go-ninja-go : unittest] 
[go-ninja-go : unittest] Previous read at 0x00c0006b1800 by goroutine 40:
[go-ninja-go : unittest]   github.com/openshift-pipelines/pipelines-as-code/pkg/adapter.listener.handleEvent.func1.1()
[go-ninja-go : unittest]       /workspace/source/pkg/adapter/adapter.go:189 +0x77
[go-ninja-go : unittest] 
[go-ninja-go : unittest] Goroutine 44 (running) created at:
[go-ninja-go : unittest]   net/http.(*Server).Serve()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:3089 +0x817
[go-ninja-go : unittest]   net/http/httptest.(*Server).goServe.func1()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/httptest/server.go:310 +0xbb
[go-ninja-go : unittest] 
[go-ninja-go : unittest] Goroutine 40 (running) created at:
[go-ninja-go : unittest]   github.com/openshift-pipelines/pipelines-as-code/pkg/adapter.listener.handleEvent.func1()
[go-ninja-go : unittest]       /workspace/source/pkg/adapter/adapter.go:188 +0x1185
[go-ninja-go : unittest]   net/http.HandlerFunc.ServeHTTP()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:2122 +0x4d
[go-ninja-go : unittest]   net/http.serverHandler.ServeHTTP()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:2936 +0x682
[go-ninja-go : unittest]   net/http.(*conn).serve()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:1995 +0xbd4
[go-ninja-go : unittest]   net/http.(*Server).Serve.func3()
[go-ninja-go : unittest]       /usr/lib/golang/src/net/http/server.go:3089 +0x58
[go-ninja-go : unittest] ==================
[go-ninja-go : unittest]     testing.go:1446: race detected during execution of test
[go-ninja-go : unittest] === RUN   TestHandleEvent/git_provider_not_detected
[go-ninja-go : unittest] === NAME  TestHandleEvent
[go-ninja-go : unittest]     testing.go:1446: race detected during execution of test
[go-ninja-go : unittest] --- FAIL: TestHandleEvent (0.02s)


```

--- a/pkg/adapter/adapter_test.go
+++ b/pkg/adapter/adapter_test.go
@@ -25,6 +25,7 @@ import (
 )
 
 func TestHandleEvent(t *testing.T) {
+	t.Parallel()
 	ctx, _ := rtesting.SetupFakeContext(t)
 	cs, _ := testclient.SeedTestData(t, ctx, testclient.Data{
 		ConfigMap: []*corev1.ConfigMap{
@@ -80,9 +81,6 @@ func TestHandleEvent(t *testing.T) {
 		},
 		logger: logger,
 	}
-
-	ts := httptest.NewServer(l.handleEvent(ctx))
-	defer ts.Close()
 
 	// valid push event
 	testEvent := github.PushEvent{Pusher: &github.User{ID: github.Int64(101)}}
@@ -143,11 +141,17 @@ func TestHandleEvent(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			req, err := http.NewRequestWithContext(context.TODO(), tt.requestType, ts.URL, bytes.NewReader(tt.event))
+			tn := tt
+			t.Parallel()
+
+			ts := httptest.NewServer(l.handleEvent(ctx))
+			defer ts.Close()
+
+			req, err := http.NewRequestWithContext(context.Background(), tn.requestType, ts.URL, bytes.NewReader(tn.event))
 			if err != nil {
 				t.Fatalf("error creating request: %s", err)
 			}
-			req.Header.Set("X-Github-Event", tt.eventType)
+			req.Header.Set("X-Github-Event", tn.eventType)
 
 			resp, err := http.DefaultClient.Do(req)
 			if err != nil {
@@ -155,8 +159,8 @@ func TestHandleEvent(t *testing.T) {
 			}
 			defer resp.Body.Close()
 
-			if resp.StatusCode != tt.statusCode {
-				t.Fatalf("expected status code : %v but got %v ", tt.statusCode, resp.StatusCode)
+			if resp.StatusCode != tn.statusCode {
+				t.Fatalf("expected status code : %v but got %v ", tn.statusCode, resp.StatusCode)
 			}
 		})
 	}

--- a/pkg/params/run.go
+++ b/pkg/params/run.go
@@ -22,7 +22,11 @@ type Run struct {
 	Info    info.Info
 }
 
+var mutex = &sync.Mutex{}
+
 func (r *Run) UpdatePACInfo(ctx context.Context) error {
+	mutex.Lock()
+	defer mutex.Unlock()
 	ns := info.GetNS(ctx)
 	if ns == "" {
 		return fmt.Errorf("failed to find namespace")

